### PR TITLE
Bug 2023356: Devfiles can't be loaded in Safari on macOS (403 - Forbidden)

### PR DESCRIPTION
**Pull request by gruselhaus**
**Fixes**: 
https://bugzilla.redhat.com/show_bug.cgi?id=2023356
https://issues.redhat.com/browse/OCPBUGSM-37199

**Analysis / Root cause**: 

In Safari 15.x on macOS, the Devfiles samples can't be loaded from the backend. As viewed in the network inspector of the browser I can say that the app makes two calls to the backend. The first answer was a redirect, the second the failure. Unfortunately, the payload gets lost between the first and second call.

**Solution Description**: 

I changed the http method from `put` to `get` and passed the `registry` via an url param instead of a payload since we are fetching data from the backend only.

**Screen shots / Gifs for design review**: 
## Before:

=?UTF-8?Q?Bildschirmfoto_2021-11-15_um_15=2E08=2E12=2Epng?=
141813321-16e7627e-198a-440b-92ab-e5850b6d3cbf

## After:
Bildschirmfoto 2021-11-15 um 19 37 20
Bildschirmfoto 2021-11-15 um 19 37 29

**Unit test coverage report**: 

**Test setup:**
Version-Release number of selected component (if applicable):
4.9

How reproducible:
Always

Steps to Reproduce:
1. Switch to dev console
2. Create a new project
3. Click on "+Add" or "View all samples"

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

**Comment by jerolimov** [comment]
/ok-to-test
/retitle Bug 2023356: Devfiles can't be loaded in Safari on macOS (403 - Forbidden)

File: pkg/server/server.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ const (
 	helmChartRepoProxyEndpoint       = "/api/helm/charts/"
 	gitopsEndpoint                   = "/api/gitops/"
 	devfileEndpoint                  = "/api/devfile/"
-	devfileSamplesEndpoint           = "/api/devfile/samples/"
+	devfileSamplesEndpoint           = "/api/devfile/samples"
```
Comment (jerolimov) [inline pkg/server/server.go:58]: This is actually the fix, but when I discussed this issue with @gruselhaus, we agreed that its not required to make a `PUT` call for fetching the devfile data. So the rest of the PR changes the `PUT` Api call to `GET`. :+1:

**Review by jerolimov — APPROVED** [review]
Verified this PR with a cluster bot instance on Chrome, Firefox and Safari (all on macOS)

Works fine :+1:

/approved
/lgtm
/retest

**Comment by jerolimov** [comment]
/cc @jhadvig @florkbr
Jakub or Bryan, can one of you please take look at the go changes? Thanks :)

**Review by jhadvig — APPROVED** [review]
/approve

**Comment by gruselhaus** [comment]
@jerolimov the `e2e-gcp-console` still fails. Is there a way to bypass this test so Tide can merge this PR?

**Pull request merged**